### PR TITLE
Dwarf Warrior (Adventure Class)

### PR DESCRIPTION
**Pull request by Osmos123**
I wanted some more flavor for dwarf adventurer specific class and more exotic type weapons such as the battle axe being used (noticed plenty of swords or battleaxes). Tried to keep the gear and stats inline with the other classes. Limited to 2 maxchosen/slots. 

It does appear when spawning in that once a slot is taken, its gone forever for the entire round. Even gibbing/killing the dwarf. I've tried spawning in many times after 2 and could not choose this class option.

https://github.com/Blackstone-SS13/BLACKSTONE/assets/126647931/ff1911fd-5c8f-4bba-b148-9030115889d3

### CODE REVIEW

File: code/modules/jobs/job_types/roguetown/adventurer/types/combat/donator/dwarfwarrior.dm [hunk 1]
```diff
@@ -0,0 +1,41 @@
+
+/datum/advclass/dwarfwarrior
+	name = "Dwarf Warrior"
```
Comment (SELFHELL) [inline code/modules/jobs/job_types/roguetown/adventurer/types/combat/donator/dwarfwarrior.dm:3]: Considering the job is already for dwarves only, looks kinda wrong to be titled a "Dwarf Dwarf Warrior"

Comment (Osmos123) [reply code/modules/jobs/job_types/roguetown/adventurer/types/combat/donator/dwarfwarrior.dm]: Thats a good point. I have changed it to just "Warrior". I think the other dwarf specific class may have the same issue and result in "Dwarf Dwarf"
image

Comment (SELFHELL) [reply code/modules/jobs/job_types/roguetown/adventurer/types/combat/donator/dwarfwarrior.dm]: I would just rename it to cavern warrior or something, theres already a warrior class; and changing the "dwarf" class would be nice

Comment (Osmos123) [reply code/modules/jobs/job_types/roguetown/adventurer/types/combat/donator/dwarfwarrior.dm]: Adjusted for both. Hopefully it will look better.

**Comment by ChungusGamer666** [comment]
Merge with main so CI checks can run

**Comment by Osmos123** [comment]
Done, hopefully I did what you were asking for.

**Comment by JTGSZ** [comment]
Looks good, why change the name of the Dwarf class to Vagrant though?

**Comment by Osmos123** [comment]
> Looks good, why change the name of the Dwarf class to Vagrant though?

I believe when a dwarf selects the dwarf class, they appear to be when examined as "Dwarf Dwarf"

image

**Comment by JTGSZ** [comment]
eh makes sense actually

**Pull request merged**